pkg/controller/migration/convert/core_test.go

```diff
@@ -90,15 +90,55 @@ var _ = Describe("core handler", func() {
 	})
 
 	Context("nodeSelector", func() {
-		It("should not set nodeSelector if none is set", func() {
-			Expect(handleCore(&comps, i)).ToNot(HaveOccurred())
-			Expect(i.Spec.ControlPlaneNodeSelector).To(BeEmpty())
+		// kube-controllers has a configurable nodeSelector which should
+		// be carried forward
+		Context("kube-controllers", func() {
+			It("should not set nodeSelector if none is set", func() {
+				Expect(handleCore(&comps, i)).ToNot(HaveOccurred())
+				Expect(i.Spec.ControlPlaneNodeSelector).To(BeEmpty())
+			})
+			It("should carry forward nodeSelector on kube-controllers", func() {
+				nodeSelector := map[string]string{"foo": "bar"}
+				comps.kubeControllers.Spec.Template.Spec.NodeSelector = nodeSelector
+				Expect(handleCore(&comps, i)).ToNot(HaveOccurred())
+				Expect(i.Spec.ControlPlaneNodeSelector).To(Equal(nodeSelector))
+			})
```

Comment:
An interesting test here would be if there is a `kubernetes.io/os: windows` nodeSelector.

Reply:
There is one! Line 122 :smile: https://github.com/tigera/operator/pull/813/files#diff-ca6adc83aab5bc88a29eb05f6f51a32aR123

Reply:
I meant on kube-controllers, I think what you linked is only done for typha and node.